proposals/NNNN-spirv-input-builtin.md

```diff
@@ -0,0 +1,139 @@
+# SPIR-V Input/Output built-ins
+
+ * Proposal: [NNNN](NNNN-spirv-input-builtin.md)
+ * Author(s): [Nathan Gauër](https://github.com/Keenuts)
+ * Status: **Design In Progress**
+
+## Introduction
+
+HLSL has semantic input/outputs parameters used to carry information
+to and from a shader, e.g: the system semantic `SV_GroupID` or `MY_SEMANTIC`,
+a user-defined semantic.
+
+In SPIR-V, those are translated to `Input` and `Output` variables,
+with either a `BuiltIn` or `Location` decoration.
+
+Input `BuiltIn` values are private to the executing lane. Reading their
+value has no side-effect. If not used, those built-in can safely be removed
+from the shader module (Unlike the inputs tagged with `Location`).
```

Comment:
In SPIR-V you can remove the user defined inputs (those with the Location decoration). Vulkan requires that the inputs are a subset of the output of the previous stage. They do not have to match exactly. You cannot remove outputs that are not written to. That will give a validation error. Here is an example where SPIR-V opt removes an unused input: https://godbolt.org/z/Mdo3WWssb.

Reply:
Right, only outputs are an issue. Removed this line, and also rephrased the one above to just say "we don't talk about Location"